### ISSUE TITLE
Chunks are marked as being written to when they're not

### ISSUE THREAD

**Issue by e-dard** [influxdata/influxdb_iox#1905]
Currently an open chunk's `time_of_last_write` is updated for every write into the chunk, even if that write failed.

https://github.com/influxdata/influxdb_iox/blob/a50c1e75e56d0f8a56696a4ffcb46cfafa252b99/server/src/db.rs#L650-L669

A chunk is movable to the Read Buffer if one of two conditions are met here:

https://github.com/influxdata/influxdb_iox/blob/a50c1e75e56d0f8a56696a4ffcb46cfafa252b99/lifecycle/src/policy.rs#L469-L497

1. The chunk's `row count` is at least `DEFAULT_MUB_ROW_THRESHOLD` (currently hard-coded to 100,000); or
2. The elapsed time between now and the chunk's most recent write is at least as large as the chunk's linger time (currently defaults to `300s`.

### Expected Behaviour

When an open chunk fails to processes new writes it can't move to the RUB based on row count so it should move to the RUB based on becoming "cold" (by no longer successfully processing writes).

TLDR; I think that `time_of_last_write` should be the _time of the last successful write_.

### Actual Behaviour

Open chunks in this state become stuck because they do not have enough rows to move to the RUB but they also are constantly treated as being written to even though the writes fail. I think one example where that could happen is with the schema validation here:

https://github.com/influxdata/influxdb_iox/blob/a50c1e75e56d0f8a56696a4ffcb46cfafa252b99/mutable_buffer/src/chunk.rs#L297-L299

I guess there could be other reasons in the future.

**Comment by e-dard** [comment]
@pauldix is going to take this one.
